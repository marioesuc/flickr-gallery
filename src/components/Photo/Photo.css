.Photo-container {
  margin: 5px;
  height: 200px;
  opacity: 0.7;
  transition: 0.4s;
}

.Photo-container:hover {
  opacity: 1;
  transition: 0.4s;
}
