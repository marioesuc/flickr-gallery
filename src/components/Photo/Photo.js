import React from 'react';
import './Photo.css';

// Custom Photo component
const Photo = props => {
  const handleOnPhotoClick = () => props.onClick(props.photoId);

  const { url, title } = props;

  return (
    <div onClick={handleOnPhotoClick}>
      <img src={url} className='Photo-container' alt={title} title={title} />
    </div>
  );
};

export default Photo;
