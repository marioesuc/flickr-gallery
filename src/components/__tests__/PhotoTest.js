import React from 'react';
import { shallow, configure } from 'enzyme';
import Photo from '../Photo/Photo';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Photo', () => {
  const props = {
    url: 'url',
    title: 'title'
  };

  it('renders correctly', () => {
    const wrapper = shallow(<Photo {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
