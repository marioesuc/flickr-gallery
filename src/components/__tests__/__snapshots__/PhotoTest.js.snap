// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo renders correctly 1`] = `ShallowWrapper {}`;
